optimum/habana/transformers/models/mixtral/modeling_mixtral.py

```diff
@@ -451,7 +451,7 @@ def gaudi_mixtral_block_dynamic_moe_forward(self, hidden_states: torch.Tensor) -
         permuted_weights=True,
         activation="silu",
         experts_min=0,
-        experts_max=7,
+        experts_max=1,
```

Comment:
Checked further and I think we should actually set the experts_max to num_experts -1.
In this case, we should set it to "len(self.experts)-1"

https://docs.habana.ai/en/latest/PyTorch/Model_Optimization_PyTorch/Custom_Ops_PyTorch.html?highlight=mixture_of_experts

It seems they hardcoded to 7 since  "num_local_experts" is 8 in the current Mixtral models.  However in UT, num_local_experts is 2, that's why we are seeing an error.